src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7053,6 +7047,55 @@ private bool IsPossibleLocalDeclarationStatement(bool isGlobalScriptLevel)
             }
         }
 
+        private bool IsPossibleTopLevelUsingLocalDeclarationStatement()
+        {
+            if (this.CurrentToken.Kind != SyntaxKind.UsingKeyword)
+            {
+                return false;
+            }
+
+            var tk = PeekToken(1).Kind;
+
+            if (tk == SyntaxKind.RefKeyword)
+            {
+                return true;
+            }
+
+            if (IsDeclarationModifier(tk)) // treat `const int x = 2;` as a local variable declaration
+            {
+                if (tk != SyntaxKind.StaticKeyword) // For `static` we still need to make sure we have a typed dentifier after it, because `using static type;` is a valid using directive.
+                {
+                    return true;
+                }
+            }
+            else if ((SyntaxFacts.IsPredefinedType(tk) &&
+                        this.PeekToken(1).Kind != SyntaxKind.DotToken && // e.g. `int.Parse()` is an expression
```

Comment:
should this use 'tk' instead?

Reply:
> should this use 'tk' instead?

Good catch, that should peek at the next token instead, i.e. 2.

---
In reply to: [392430189](https://github.com/dotnet/roslyn/pull/42385#discussion_r392430189) [](ancestors = 392430189)